### PR TITLE
docs: frontend

### PR DESCRIPTION
**Pull request by storyworks**
# Description

Description of our frontend project, how and when to write stories.

There are probably more sections needed. I will add in next PRs

# How should this PR be QA Tested?

- [ ] Checks are green
- [ ] Documentation makes sense

# Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged into main

### CODE REVIEW

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+Currently we have a 80k snapshot limit per month for visual regression testing.
+
+## What is a story?
+
+[A rendered state of a UI component](https://storybook.js.org/docs/react/get-started/whats-a-story)
+
+## When to write stories
+
+For instances of mui components we directly use, a `components.stories.tsx` story should be included in the shared-ui theme tokens folder. <br/>
+[journeys]() - to be created <br/>
```
Comment (tataihono) [inline apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:21]: not having a link is causing a build error.

```
Error: Markdown link URL is mandatory in "apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md" file (title: component types, line: 20).
```

**Review by tataihono — DISMISSED** [review]

File: apps/docs/docs/03-basics/03-frontend/02-component-library/index.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+All Material UI components will be themed with the `mui` [default theme](https://mui.com/material-ui/customization/default-theme/).
+
+Since we reuse these high level `mui` components to build many different components, these are known as **common** components.
+
+### Our components
+
+We need the Mui components to look and/or function differently to the default. We apply multiple different strategies for each use case.
+
+For styling & function changes for **all** components:
+![All components](./all-components.png)
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/02-component-library/index.md:38]: If the structure of the monorep is set, I wonder if it could be a good idea to have the pathname written down as well along with the diagram.

e.g /core/libs/shared/ui/src/libs/themes/base/tokens

It took me awhile to understand what "base" meant and what theme tokens were. I found it helpful to be able to follow along in the monorepo.

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+For instances of mui components we directly use, a `components.stories.tsx` story should be included in the shared-ui theme tokens folder. <br/>
+journeys - to be created <br/>
+[watch](https://storybook.core.jesusfilm.org/?path=/story/website-theme--components)
+
+For our own components use the following chart:
+
+![When to write stories](./when-to-write-stories.png)
+
+> Not sure how the component is made? See [component types](#component-types)
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:28]: This points me to https://github.com/JesusFilm/core/pull/1053/files#component-types

Is this intended?

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+
+In our monorepo we can categorize all the components we create into **common** and **regular** components, of which there are several types:
+
+**Common components**<br/>
+Any component intended for reuse in multiple different components. This includes:
+
+- Some unique component made from multiple components like
+  > For example [journeys-admin Button](https://storybook.core.jesusfilm.org/?path=/story/journeys-admin-editor-controlpanel-button--default) is made from Mui-Box, Mui-Typography and Mui-Stack amongst others.
+- An extension of a component.
+  > For example [journeys-ui TextField](https://storybook.core.jesusfilm.org/?path=/story/journeys-ui-textfield--states) uses the `filled` variant of [Mui-TextField](https://mui.com/material-ui/react-text-field/#form-props) and adapts it to work with formik - a form managagement library.
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:100]:
```suggestion
  > For example [journeys-ui TextField](https://storybook.core.jesusfilm.org/?path=/story/journeys-ui-textfield--states) uses the `filled` variant of [Mui-TextField](https://mui.com/material-ui/react-text-field/#form-props) and adapts it to work with formik - a form management library.
```

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+
+- Each story should use it's project-specific config to correctly set the theme. If the component is simple and does not change based on viewport, use the project-specific `simpleComponentConfig`.
+
+- Our project uses `ComponentStory<typeof Component>` as the Template type. If you need to extend the template type, use
+
+```
+Story< Parameters<typeof Component>[0] & { newProperty: NewPropertyType }
+```
+
+## Component Types
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:90]: Should the explanation about component types come before Stories? 

when-to-write-stories.png uses language such as "regular" and "common" components without explanation.

File: apps/docs/docs/03-basics/03-frontend/02-component-library/index.md (hunk 2 above)
Comment (storyworks) [reply apps/docs/docs/03-basics/03-frontend/02-component-library/index.md:38]: Great idea!

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md (hunk 3 above)
Comment (storyworks) [reply apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:28]: hmm yeah it's supposed to take you to the Components type section - will fix

File: apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md (hunk 5 above)
Comment (storyworks) [reply apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:90]: I did consider that - however, I expect people to look at the component types section once and after they understand they won't need to reference it.. So didn't want it to be at the top of a page. Maybe I should move it back to the bottom of the components-library doc page? Assuming people read that page first.

Comment (edmonday) [reply apps/docs/docs/03-basics/03-frontend/04-visual-testing/index.md:90]: Just leave it as is, was not aware that this was mentioned previously in another page.

File: apps/docs/docs/03-basics/03-frontend/04-storybook/index.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+3. Some components call **external** apis. For this we use `msw`. We define the mock in a `.handlers.ts` file. See `VideoFromYouTube.handlers.ts` & `VideoFromYoutube.stories.tsx` for a usage example.
+
+4. Instead of bloating the Template or story args, we extract any complex component data. This may be defined above the Template or imported from a `data.ts` file within the same component folder. See `RadioQuestion.stories.tsx` for an example.
+
+5. Our project uses `ComponentStory<typeof Component>` as the Template type. If you need to extend the template type, use
+
+```
+Story< Parameters<typeof Component>[0] & { newProperty: NewPropertyType }
+```
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/04-storybook/index.md:115]: I didn't quite understand this, is there a link that would help with understanding this?

`Story< Parameters<typeof Component>[0] & { newProperty: NewPropertyType }`

This line was hard to follow, does it require further explanation? Why is there a [0] & ?

File: apps/docs/docs/03-basics/03-frontend/02-component-library/index.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+**UI Libraries**
+
+- [journey-ui](https://storybook.core.jesusfilm.org/?path=/story/journeys-ui-card--default): common components used in `journeys` & `journeys-admin`
+- [shared-ui](https://storybook.core.jesusfilm.org/?path=/story/shared-ui-dialog--basic): common components used in all app projects
+
+**Themes**
+
+- [base](https://storybook.core.jesusfilm.org/?path=/story/default-theme--viewport): default theme for Next Steps. Includes project specific styles for journeys.
+- [website](https://storybook.core.jesusfilm.org/?path=/story/website-theme--components): project theme for Jesus Film Project websites
+- [admin](https://storybook.core.jesusfilm.org/?path=/story/admin-theme--colors): project theme for admin projects.
```
Comment (edmonday) [inline apps/docs/docs/03-basics/03-frontend/02-component-library/index.md:62]: should these all point to the same page for each theme? For example the color pages? Was confusing for "base" when all I saw was a black screen in the story.

File: apps/docs/docs/03-basics/03-frontend/02-component-library/index.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+  - [watch](https://storybook.core.jesusfilm.org/?path=/story/watch-pagewrapper--default)
+
+**UI Libraries**
+
+- [journey-ui](https://storybook.core.jesusfilm.org/?path=/story/journeys-ui-card--default): common components used in `journeys` & `journeys-admin`
+- [shared-ui](https://storybook.core.jesusfilm.org/?path=/story/shared-ui-dialog--basic): common components used in all app projects
+
+**Themes**
+
+- [base](https://storybook.core.jesusfilm.org/?path=/story/default-theme--viewport): default theme for Next Steps. Includes project specific styles for journeys.
```
[the remark this answers is not in the document]
Comment (storyworks) [reply apps/docs/docs/03-basics/03-frontend/02-component-library/index.md:60]: Good question, I wasn't 100% sure - I pointed it to the pages with the most distinguishing features (ie in the future base might just be breakpoints and we seperate out journeys theme into it's own theme..)

Can do colours though .

File: apps/docs/docs/03-basics/03-frontend/04-storybook/index.md (hunk 6 above)
Comment (storyworks) [reply apps/docs/docs/03-basics/03-frontend/04-storybook/index.md:115]: https://www.typescriptlang.org/docs/handbook/utility-types.html#parameterstype We were using this which returns an array with 1 object so had to use `[0]` to get it. However I will update this to use 

`StoryObj<Component>` instead

File: apps/docs/docs/03-basics/03-frontend/02-component-library/index.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (edmonday) [reply apps/docs/docs/03-basics/03-frontend/02-component-library/index.md:60]: Oh yup algood! I think initially it was confusing since it took me to the viewport page and it was just a black screen.

**Review by edmonday — APPROVED** [review]
